### ISSUE TITLE
logit转prob应该是用softmax?

### ISSUE THREAD

**Issue by zenghsh3** [PaddlePaddle/Paddle#43971]
### 问题描述 Please describe your issue

https://github.com/PaddlePaddle/Paddle/blob/a072fca8229b26042fe24bff42989533e1d2050a/python/paddle/distribution/categorical.py#L119

Pytorch参考：
https://github.com/pytorch/pytorch/blob/0a1bc5f501bb571b6f4275b6ca863e68bf6cda02/torch/distributions/utils.py#L65

**Comment by 0x45f** [comment]
你好，可以尝试使用paddle中的对应api组出logits_to_probs函数，可以参考下面的代码：
![image](https://user-images.githubusercontent.com/23097963/176627735-d0d0d463-c632-4c76-b981-4c589c087115.png)

**Comment by zenghsh3** [comment]
主要是想反馈Categorical分布计算probs的方式应该不正确

**Comment by pangyoki** [comment]
> 主要是想反馈Categorical分布计算probs的方式应该不正确

你好，请问有具体例子吗？使用paddle和torch得到的结果不一样吗？我理解你对比的paddle和torch的这两块代码讲的不是同一个事情。paddle这个probs是和log_probs对应的接口，需要传入输入tensor的。torch好像是没有对应接口的。

**Comment by zenghsh3** [comment]
https://github.com/pytorch/pytorch/blob/c980fc3d3c8db5bff6fd10dd746dc56c55b82fcd/torch/distributions/categorical.py#L95
这是torch的categorical分布计算prob

**Comment by zenghsh3** [comment]
例如神经网络输出的logit为[-1, 0]，paddle方式算出的概率为[1, 0]，但torch算出的概率是[0.27, 0.73]左右

**Comment by cxxly** [comment]
这是一个历史遗留问题，Categorical构造参数虽然命名为logits，但是其中prob方法把logits当作概率 'probs' 进行计算, sample方法用作logits进行计算。 `self._prob = self.logits / dist_sum` 这行代码实际上是吧 logits当作概率, 除以dist_sum是为了保证概率之和为1。
 非常感谢您的反馈，由于是不兼容升级，所以暂时没有修改。辛苦反馈给产品进行排期 @0x45f

**Comment by zenghsh3** [comment]
好的，还是要尽快修复下，容易踩坑
